doc/decisions/0012-repeating-answer-options.md

```diff
@@ -0,0 +1,274 @@
+- `dynamic_answers` will be limited to a single block i.e. they will only be displayed on one page.
+
+#### `dynamic_answers.values`
+
+- This field will contain the `source` for the list of literal values to be repeated over.
+- The `values` property will follow existing patterns and allow either a `Value Source` or a `Value Operator` as per the schema examples below.
+- There must be at least one item in the list being used to generate dynamic answers unless static answers are provided.
+
+#### `dynamic_answers.answers`
+- This property allows a list of answer blocks, as there can be more than one dynamic answer for each literal list item.
+- The `answers` property should be able to support any of the properties that would be supported in a normal/static `answer` block.
+- All the properties within the `dynamic_answers.answers` block that can contain text displayed to the user will need to support placeholder transforms. For example, answer labels which currently do not allow placeholders will need to be extended to support this.
+- `dynamic_options` cannot be used inside `dynamic_answers`.
+- The structure should be consistent with a regular static answer block.
+- For `dynamic_answers.answers` blocks, the answer `id` will simply be a prefix, and a unique identifier will need to be added to the prefix at runtime. This is required in order for us to be able to render the id in other parts of the questionnaire. This id made of the prefix and unique identifier will only be used for front end rendering purposes. We will still use just the `list_item_id` for storage even if it isn't a list collector.
+
+### Dynamic answers based on a list source
+
+```json
+{
+  "question": {
+    "id": "percentage-of-research",
+    "title": "Some title",
+    "type": "Question",
+    "answers": [...],
+    "dynamic_answers": {
+      "values": {
+        "source": "list",
+        "identifier": "postcodes",
+        "selector": "items"
+      },
+      "answers": [
+        {
+          "description": {
+            "text": "Percentage of shopping at {transformed_value}",
+            "placeholders": [
+              {
+                "placeholder": "transformed_value",
+                "value": {
+                  "source": "answers",
+                  "identifier": "postcode"
+                }
+              }
+            ]
+          },
+          "id": "percentage-of-shopping",
+          "mandatory": false,
+          "type": "Percentage",
+          "maximum": {
+            "value": 100
+          },
+          "decimal_places": 2
+        }
+      ]
+    }
+  }
+}
+```
+- For dynamic answers driven by list collectors, for front end purposes each list item, at runtime the `list-item-id` or `selector` will be appended to the `id` used in the `dynamic_answers.answer` block, so that all answers have unique ids e.g. `"id": "percentage-of-shopping-{list_item_item_id}`. For storage we will just use the `list_item_id`.
```

Comment:
```suggestion
- For dynamic answers driven by list collectors, for front end purposes each list item, at runtime the `list-item-id` will be appended to the `id` used in the `dynamic_answers.answer` block, so that all answers have unique ids e.g. `"id": "percentage-of-shopping-{list_item_id}`. For storage we will just use the combination of `answer_id` and `list_item_id`.
```